src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -97,6 +97,37 @@ public static SourceText From(string text, Encoding encoding = null, SourceHashA
             return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
         }
 
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
```

Comment:
will do. I just wanted quick check from you guys whether this is even okay to do.

is checksum and the embedded blob thing not there okay? I think I am using encoding wrong though. since I am getting TextReader, it should be already normalized to certain encoding, the given encoding should be just passed along like StringText.From(string, encoding), and not actually used anywhere else.